### DISCUSSION TITLE
Support for Virtual IP advertisement from  VM level

### DISCUSSION THREAD

**Discussion by ganesh-karthick** [projectcalico/calico#8395] [Q&A]
Hi,

Based on [documentation](https://docs.tigera.io/calico/latest/networking/configuring/advertise-service-ips), I see calico has support for  advertising IP's from service(external/cluster), LB oriented services . However, I have a need to advertise a Virtual IP on  specific interfaces at a VM level which is running kubernetes. The Virtual IP is maintained via  VRRP outside of kube system. Is there any way to ask to calico to advertise specific IP provided it exists on a VM in an interface via BGP , kind of like static routes in BGP. I could do this natively via Bird, was wondering if there is equivalent in calico

Example bird syntax

```
# Static routes (Again, there can be multiple instances, for different address
# families and to disable/enable various groups of static routes on the fly).
protocol static {
    ipv4;            # Again, IPv4 channel with default options
    route some_vip1/some_mask via some_ip;
    route some_vip2/some_mask via some_ip;
}
```

**Comment by caseydavenport** [comment]
@ganesh-karthick sorry for the delay!

Right now, there's no way to specify a static route explicitly for Calico to advertise. However, Calico does read routes from the host and will export any routes it learns from the host that belong to a Calico IP pool.

One thing you can try is to create an IP pool that encompasses the Virtual IPs that you would like exported - Calico should then export those routes from the host's that own the IPs.

You will want to make sure that the IP pool is marked as "disabled"  to make sure that Calico doesn't attempt to assign pod IPs from that pool.

For example:

```yaml
kind: IPPool
apiVersion: projectcalico.org/v3
metadata:
  name: virtual-ip-range
spec:
  disabled: true
  cidr: X.X.X.X
```

**Reply by ganesh-karthick** [reply]
Thanks, will try this out. Are there any future plans to make static routes calico first class block in yaml definition , so its easier to use?

**Reply by caseydavenport** [reply]
It's something that has come up in the past, but I don't think it's on the immediate roadmap. Not opposed to adding such a feature, it just hasn't seen a lot of interest in recent times.

**Reply by ganesh-karthick** [reply]
Thanks, in future , if we could add this support it would be great  for virtual IP users at VM level who also use calico as CNI.
